Clean up grammar in README

kevinburkeshyp opened the pull request:
Makes some of the sentences read more like idiomatic English. Try to document
some of the function signatures in the headers. Add syntax highlighting for two
of the examples.

kevinburke committed:
Clean up grammar in README

Makes some of the sentences read more like idiomatic English. Try to document
some of the function signatures in the headers. Add syntax highlighting for two
of the examples.

kevinburkeshyp commented:
In addition - is this actually an assertion? `promised('abc').should.finally.be.a.String`. Should that be `promised('abc').should.finally.be.a(String)`

btd commented:
Hi, thanks for fixes.
`promised('abc').should.finally.be.a.String` is not an assertion because `.String` method not called. It should be `.String()`. Your second version will not work with should.js (that old syntax was removed in initial version - it will work maybe only in chai).
Can you pls also fix title for `.rejectedWith(error)`?

kevinburke committed:
Call assertion functions in README docs

The previous version simply accessed the property, it didn't trigger the
assertion. Add parentheses to trigger the assertion.

benbuckman commented on README.md:
```diff
@@ -39,13 +39,13 @@ it('should be allow to check if promise rejected', function() {
 
 ## .rejectedWith
```
Suggestion, `.rejectedWith(err)` to match the `.fulfilledWith(value)` change above?
Thanks

benbuckman replied:
Actually that's not the real signature. It's confusing – maybe best to just link to it – http://shouldjs.github.io/#assertion-throw

kevinburkeshyp commented:
Pushed fixes in 2ebbcbe.

benbuckman commented:
:+1:

btd commented:
Thanks, guys :+1:
